### PR TITLE
[fix] layout converter fix

### PR DESCRIPTION
**Pull request by graceguo-supercat**
- small bug fix for position data clean-up function.
- add changed_on data into initial sliceEntities (used for sortby most recent)
- add unit tests for SliceAdder component

@williaster

### CODE REVIEW

File: superset/assets/spec/javascripts/dashboard/fixtures.jsx [hunk 1]
```diff
@@ -169,3 +169,141 @@ export const {
   datasources: {},
   user_id: '1',
 });
+
+export const mockSliceEntities = {
```
Comment (williaster) [inline superset/assets/spec/javascripts/dashboard/fixtures.jsx]: @graceguo-supercat can we actually delete this file? all of the fixtures for the new dashboard are within `spec/javascripts/dashboard/fixtures/` and are separated by type (not all in one file). there's actually one for `mockSliceEntities` but you could add to it.

File: superset/assets/src/dashboard/util/propShapes.jsx [hunk 2]
```diff
@@ -59,8 +59,8 @@ export const slicePropShape = PropTypes.shape({
   datasource: PropTypes.string,
   datasource_name: PropTypes.string,
   datasource_link: PropTypes.string,
-  changedOn: PropTypes.number,
-  modified: PropTypes.string,
+  changed_on: PropTypes.number.isRequired,
+  modified: PropTypes.string.isRequired,
```
Comment (williaster) [inline superset/assets/src/dashboard/util/propShapes.jsx]: out of curiosity, why are these both not the same type? that seems more ideal.

**Review by williaster — COMMENTED** [review]
this lgtm but would like to consolidate the fixtures into the new directory!

File: superset/assets/src/dashboard/util/propShapes.jsx (hunk 2 above)
Comment (graceguo-supercat) [reply superset/assets/src/dashboard/util/propShapes.jsx]: changed_on is timestamp number, modified is human readable string like `a month ago`, which is server-side provided convert timestamp to readable text.

**Comment by graceguo-supercat** [comment]
i removed old fixtures.js, seems not in-use by any unit test.

File: superset/assets/spec/javascripts/dashboard/fixtures/mockSliceEntities.js [hunk 3]
```diff
@@ -37,3 +37,141 @@ export default {
   errorMessage: null,
   lastUpdated: 0,
 };
+
+export const sliceEntitiesForDashboard = {
+  slices: {
+    '127': {
```
Comment (williaster) [inline superset/assets/spec/javascripts/dashboard/fixtures/mockSliceEntities.js]: nit -- these don't need quotes / looks like prettier doesn't cover fixtures. nbd we can fix in another linting pass.

**Review by williaster — APPROVED** [review]
LGTM

**Pull request merged**